index.d.ts

```diff
@@ -1,66 +1,257 @@
-export type WidgetId = string;
-export type Container = ElementId | HTMLElement;
-
-declare global {
-	interface Window {
-		turnstile: TurnstileObject;
-	}
-}
+/**
+ * Represents a rendered Turnstile widget.  Used to identify a specific widget when calling
+ * Turnstile methods.
+ */
+export type WidgetId = string
 
+/**
+ * Turnstile is Cloudflare’s smart CAPTCHA alternative. It can be embedded into any website
+ * without sending traffic through Cloudflare and works without showing visitors a CAPTCHA.
+ * @see https://developers.cloudflare.com/turnstile
+ */
 export interface TurnstileObject {
-	ready: (cb: () => any) => void;
-	implicitRender: () => void;
-	execute: (container?: Container, options?: TurnstileOptions) => void;
-	render: (container: Container, options: TurnstileOptions) => WidgetId;
-	reset: (widget?: Container | WidgetId) => void;
-	remove: (widget?: Container | WidgetId) => void;
-	getResponse: (widget?: Container | WidgetId) => string | undefined;
-	isExpired: (widget?: Container | WidgetId) => boolean;
+	/**
+	 * If using synchronous loading, will be called once the DOM is ready.
+	 * @see [Explicit Rendering](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#explicitly-render-the-turnstile-widget)
+	 */
+	ready: (cb: () => any) => void
+
+	/**
+	 * @see [Implicit Rendering](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#implicitly-render-the-turnstile-widget)
+	 */
+	implicitRender: () => void
+
+	/**
+	 * Excecutes the challenge after the render() function has been called, by invoking the
+	 * turnstile.execute(container: string | HTMLElement, jsParams?: RenderParameters) function
+	 * separately, decoupling the appearance and rendering of a widget from its execution.
+	 * @see [Execution Modes](https://developers.cloudflare.com/turnstile/get-started/client-side-rendering/#execution-modes)
+	 */
+	execute: () => void
+
+	/**
+	 * Explicitly renders the Turnstile widget.  The `sitekey` parameter is required. The
+	 * `callback` parameter, however, is only required if `execution` is set to `execute`.
+	 * @param container - HTML container to render the widget.
+	 * @param options - Configuration options for rendering. See {@link RenderOptions}.
+	 * @returns A `widgetId` string if successful, otherwise undefined.
+	 */
+	render(container: HTMLElement | string, options: RenderOptions): string | undefined
+
+	/**
+	 * Resets the widget.
+	 * @param widgetId - The ID of the widget.
+	 */
+	reset(widgetId: WidgetId): void
+
+	/**
+	 * Removes the widget from the page.
+	 * @param widgetId - The ID of the widget.
+	 */
+	remove(widgetId: WidgetId): void
+
+	/**
+	 * Obtains the widget's response using its widgetId.
+	 * @param widgetId - The ID of the widget.
+	 */
+	getResponse(widgetId: WidgetId): string
+
+	/**
+	 * Checks if the widget has expired.
+	 * @param widgetId - The ID of the widget.
+	 */
+	isExpired(widgetId: WidgetId): boolean
```

Comment:
This also accepts a HTMLElement or element id (previously expressed as `Container` type)